src/components/inputs/select-input/select-input.js

```diff
@@ -15,277 +15,255 @@ import DropdownIndicator from '../../internals/dropdown-indicator';
-    menuShouldBlockScroll: PropTypes.bool,
-    // menuShouldScrollIntoView: PropTypes.bool,
-    name: PropTypes.string,
-    noOptionsMessage: PropTypes.func,
-    onBlur: PropTypes.func,
-    onChange: PropTypes.func,
-    onFocus: PropTypes.func,
-    onInputChange: PropTypes.func,
-    // onKeyDown: PropTypes.func,
-    // onMenuOpen: PropTypes.func,
-    // onMenuClose: PropTypes.func,
-    // onMenuScrollToTop: PropTypes.func,
-    // onMenuScrollToBottom: PropTypes.func,
-    // openMenuOnFocus: PropTypes.bool,
-    // openMenuOnClick: PropTypes.bool,
-    options: PropTypes.arrayOf(
-      PropTypes.oneOfType([
-        PropTypes.shape({ value: PropTypes.string.isRequired }),
-        PropTypes.shape({
-          options: PropTypes.arrayOf(
-            PropTypes.shape({ value: PropTypes.string.isRequired })
-          ),
-        }),
-      ])
-    ),
-    showOptionGroupDivider: PropTypes.bool,
-    // pageSize: PropTypes.number,
-    placeholder: PropTypes.string,
-    // screenReaderStatus: PropTypes.func,
-    // styles: PropTypes.objectOf(PropTypes.func),
-    // theme: PropTypes.object,
-    tabIndex: PropTypes.string,
-    tabSelectsValue: PropTypes.bool,
-    value: (props, ...rest) =>
-      props.isMulti
-        ? PropTypes.arrayOf(PropTypes.string).isRequired(props, ...rest)
-        : PropTypes.string(props, ...rest),
-  };
+  const placeholder =
+    props.placeholder || intl.formatMessage(messages.placeholder);
+  // Options can be grouped as
+  //   const colourOptions = [{ value: 'green', label: 'Green' }];
+  //   const flavourOptions = [{ value: 'vanilla', label: 'Vanilla' }];
+  //   const groupedOptions = [
+  //     { label: 'Colours', options: colourOptions },
+  //     { label: 'Flavours', options: flavourOptions },
+  //   ];
+  // So we "ungroup" the options by merging them all into one list first.
+  const optionsWithoutGroups = flatMap(props.options, option =>
+    has(option, 'value') ? option : option.options
+  );
 
-  state = { selectedOptions: [] };
+  const selectedOptions = props.isMulti
+    ? props.value
+        // Pass the options in the order selected by the use, so that the
+        // sorting is not lost
+        .map(value =>
+          optionsWithoutGroups.find(option => option.value === value)
+        )
+        .filter(Boolean)
+    : optionsWithoutGroups.find(
+        option => has(option, 'value') && option.value === props.value
+      ) || null;
 
-  static getDerivedStateFromProps = props => {
-    // Options can be grouped as
-    //   const colourOptions = [{ value: 'green', label: 'Green' }];
-    //   const flavourOptions = [{ value: 'vanilla', label: 'Vanilla' }];
-    //   const groupedOptions = [
-    //     { label: 'Colours', options: colourOptions },
-    //     { label: 'Flavours', options: flavourOptions },
-    //   ];
-    // So we "ungroup" the options by merging them all into one list first.
-    const optionsWithoutGroups = flatMap(props.options, option =>
```

Comment:
In L40 this has been replaced without using state. Why was this in a gDSFP before?

Reply:
I'm not really sure why it was there in the first place. We were using a gDSFP without a condition, so it always updated "state" to the new value on every render (including first render), and then we never set the state or did anything with it. Same thing in the other components.

Reply:
Odd. We might just have adopted it cause it was the new thing.

Reply:
It's possible that our first implementation of the component had setState happening but then we refactored it out but left the gDSFP 🤷‍♂ ?